src/openforms/tests/e2e/test_input_validation.py

```diff
@@ -292,3 +317,209 @@ def test_max_date_fixed_value(self):
+            "validate": {"max": 15},
+        }
+
+        self.assertValidationIsAligned(
+            component,
+            ui_input=50,
+            expected_ui_error="De waarde moet 15 of kleiner zijn.",
+        )
+
+
+class SingleMapTests(ValidationsTestCase):
+    async def apply_ui_input(self, page: Page, label: str, ui_input: str | int | float):
+        page.wait_for_selector(
+            ".openforms-leaflet-map, [aria-label='Required map']", state="visible"
+        )
+
+    def test_required_field(self):
+        component: Component = {
+            "type": "map",
+            "key": "requiredMap",
+            "label": "Required map",
+            "validate": {"required": True},
+        }
+
+        self.assertValidationIsAligned(
+            component,
+            ui_input="",
+            expected_ui_error="Het verplichte veld Required map is niet ingevuld.",
+        )
+
+
+class SinglePostcodeTests(ValidationsTestCase):
+    def test_required_field(self):
+        component: Component = {
+            "type": "postcode",
+            "key": "requiredPostcode",
+            "label": "Required postcode",
+            "validate": {"required": True},
+        }
+
+        self.assertValidationIsAligned(
+            component,
+            ui_input="",
+            expected_ui_error="Het verplichte veld Required postcode is niet ingevuld.",
+        )
+
+
+class SingleSignatureTests(ValidationsTestCase):
+
+    async def apply_ui_input(self, page: Page, label: str, ui_input: str | int | float):
+        page.wait_for_selector("[aria-label='Required signature']", state="visible")
+
+    def test_required_field(self):
+        component: Component = {
+            "type": "signature",
+            "key": "requiredSignature",
+            "label": "Required signature",
+            "validate": {"required": True},
+        }
+
+        self.assertValidationIsAligned(
+            component,
+            ui_input="",
+            expected_ui_error="Het verplichte veld Required signature is niet ingevuld.",
+        )
+
+
+class SingleTextAreaTests(ValidationsTestCase):
+    def test_required_field(self):
+        component: Component = {
+            "type": "textarea",
+            "key": "requiredTextarea",
+            "label": "Required textarea",
+            "validate": {"required": True},
+        }
+
+        self.assertValidationIsAligned(
+            component,
+            ui_input="",
+            expected_ui_error="Het verplichte veld Required textarea is niet ingevuld.",
+        )
+
+    def test_max_length(self):
+        component: Component = {
+            "type": "textarea",
+            "key": "requiredTextarea",
+            "label": "Required textarea",
+            "validate": {"maxLength": 10},
+        }
+        invalid_sample = "word" * 4
+
+        self.assertValidationIsAligned(
+            component,
+            ui_input=invalid_sample,
+            api_value=invalid_sample,
```

Comment:
```suggestion
            ui_input=invalid_sample,
```
if they're the same, don't specify `api_value` :) the test is more robust by using `ui_input` for both inputs (UI + API)